tiny_dnn/core/backend.h

```diff
@@ -62,9 +62,12 @@ inline backend_t default_engine() {
 #ifdef CNN_USE_AVX
 #if defined(__AVX__) || defined(__AVX2__)
     return backend_t::avx;
+#else
+#error "your compiler does not support AVX"
 #endif
-#endif // CNN_USE_AVX
+#else
     return backend_t::internal;
+#endif
 }
```

Comment:
The reason for the change in `default_engine` function is Visual Studio emits `warning C4702: unreachable code` with Warning Level4 option `/W4`.

Compiler flags for AVX are checked in `CMakeLists.txt` so another check with preprocessor macro can be a bit redundant but I think it's not harmful.

Using AVX backend as a default backend sounds a bit unnatural because it isn't a reference implementation.